risc0/zkvm/src/host/server/exec/executor.rs

```diff
@@ -771,3 +776,58 @@ impl<'a> ExecutorImpl<'a> {
         ))
     }
 }
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExecutorStats {
+    execution_time: Duration,
+    segment_count: usize,
+    total_prover_cycles: u64,
+    user_instruction_cycles: u64,
+    cycles_used_ratio: f64,
+}
+
+impl ExecutorStats {
+    fn new(session: &Session, elapsed: Duration) -> Result<Self> {
+        let (total_prover_cycles, user_instruction_cycles) = session.get_cycles()?;
+        let cycles_used_ratio = user_instruction_cycles as f64 / total_prover_cycles as f64 * 100.0;
+        let segment_count = session.resolve()?.len();
```

Comment:
Resolve is expensive, we should not call this here

Reply:
how about `session.segments.len()` instead?

Reply:
Oh, thanks for pointing this out!

Would it be okay to call `session.segments.len()`? I was thinking that, since `resolve` is a fallible call, it might be the case that the real `segment_count` might not be exactly the same as `session.segments.len()` if it happens to be the case that the `resolve()` call fails?

Looking into the implementation to `resolve()` it seems like it's not the case, and `session.segments.len()` should always be valid - but would like to confirm :)

Reply:
`session.segments.len()` is the right thing to use here.